### PR TITLE
Base of the sunburst chart, make color functions a little more sane and ...

### PR DESCRIPTION
**Pull request by liquidpele**
...have ability to use them as a normal ordinal scale too.

https://github.com/novus/nvd3/issues/781

### CODE REVIEW

**Comment by liquidpele** [comment]
This is pretty basic, the only change that would affect anythign else is with the color functions.  Looking into things more, it looks like you're *supposed* to use an ordinal scale for colors, instead of the (d, i) functions that we've been using, and the sunburst was doing so based on the name of the data instead of any kind of index, so I modified the functions to just use a scale if i isn't passed to the function (meaning get the color by value instead of by index).

File: src/models/sunburst.js [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+                        x.domain(xd(t));
+                        y.domain(yd(t)).range(yr(t));
+                        return arc(d);
+                    };
+                };
+            }
+
+        });
+
+        renderWatch.renderEnd('pie immediate');
```
Comment (robinfhu) [inline src/models/sunburst.js]: should probably say 'sunburst immediate' just for consistency

Comment (liquidpele) [inline src/models/sunburst.js]: doh

**Comment by robinfhu** [comment]
This is pretty amazing work @liquidpele .
sunburst

Since it doesn't really affect other charts go ahead an add it.  Also see if you can add a unit test that renders the whole chart, so we get some code coverage on it.  Really cool.

**Comment by robinfhu** [comment]
BTW @liquidpele , have you used the code coverage tool built into the nvd3 build process?
code coverage

You can see it by going to the 'coverage/' folder after running 'grunt'.

**Comment by liquidpele** [comment]
Added some unit tests so it has some level of coverage per robin, merging now.

**Pull request merged**